Add more field to HCR_EL2

kisum9 opened the pull request:
Add definitions for the HCR_EL2.TVM and HCR_EL2.VSE fields, corresponding to bit [26] ("TVM") and bit [8] ("VSE") as defined in the Arm Architecture Reference Manual.

Commit:
Add more field to HCR_EL2

VSE, TVM

Signed-off-by: Yan Tan <tanyan@kylinos.cn>

jonathanpallant commented:
Please check the rendered output from running `cargo doc`. You have some bullet lists without bullets that I believe will render poorly.

kisum9 commented:
> Please check the rendered output from running `cargo doc`. You have some bullet lists without bullets that I believe will render poorly.

Thanks for pointing that out! I've fixed the issues you mentioned and updated the PR. Please take another look when you have time.

berkus commented:
<img width="916" height="109" alt="image" src="https://github.com/user-attachments/assets/b31c64dc-2c11-4bff-97fd-f4083078aa4c" />

There are still some broken doc lines in the crate level documentation. I assume the other PRs also have similar issues?

berkus reviewed (changes requested):
Please fix them.

kisum9 commented:
> Please fix them.

It seems this is a pre-existing issue. Would you prefer I fix all the HCR_EL2 broken doc lines in this PR？

kisum9 commented:
<img width="1872" height="150" alt="image" src="https://github.com/user-attachments/assets/911b2c64-e914-4842-a31b-eb7b3e02c0d1" />
<img width="2126" height="128" alt="image" src="https://github.com/user-attachments/assets/268e5902-50de-4300-ba86-93651043fcf9" />
<img width="1676" height="98" alt="image" src="https://github.com/user-attachments/assets/bc145151-5926-49a6-b25a-cbeffbea6321" />
and others

berkus commented:
Yep, just moving them apart from the first line should fix the issue.

kisum9 commented:
> Yep, just moving them apart from the first line should fix the issue.

I’ve fixed the issue. take another look when you have time, thanks ^^

berkus commented:
> I’ve fixed the issue. take another look when you have time, thanks ^^

Thank you!

kisum9 commented:
Happy to help :)
